src/components/Header/Header.js

```diff
@@ -1,17 +1,24 @@
+import { faMoon, faSun } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import React from 'react'
 import { Link } from 'react-router-dom'
+import { useContext } from 'react'
+
 import './Header.css'
+import  { ThemeContext } from '../../Context/ThemeContextProvider'
 
 const Header = () => {
+  const {theme,ToggleMode}=useContext(ThemeContext)
   return (
-    <div class="navigation-container flex-hz">
+    <div className={theme=="light"?"navigation-container flex-hz":"navigation-container-dark flex-hz"}>
```

Comment:
It's not a wrong way to add class but you can consider to make a string template then add the class which is needed in both dark and light theme and then use ternary operator to add specific classes regarding theme. Here you are adding flex-hz two times which is common in both theme